challenge_help_me_carry/src/challenge_hmc_functions/hmc_functions.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+import roslib;
+import rospy
+import smach
+import sys
+import time
+import datetime
+from hmi import TimeoutException
+
+from cb_planner_msgs_srvs.msg import PositionConstraint
+
+import robot_smach_states.util.designators as ds
+from robot_smach_states.util.designators import check_type
+from robot_skills.arms import Arm, GripperState
+
+import robot_smach_states as states
+
+from robocup_knowledge import load_knowledge
+from robot_skills.util import kdl_conversions
+from robot_skills.util.entity import Entity
+
+challenge_knowledge = load_knowledge('challenge_help_me_carry')
+
+def setup_challenge(self, robot):
+
+    self.place_name = ds.EntityByIdDesignator(robot, id=challenge_knowledge.default_place, name="place_name")
+    self.place_position = ds.LockingDesignator(ds.EmptySpotDesignator(robot, self.place_name, 
+                                                                    name="placement",
+                                                                    area=challenge_knowledge.default_area),
+                                                                    name="place_position")
+
+    self.empty_arm_designator = ds.UnoccupiedArmDesignator(robot.arms, robot.rightArm, name="empty_arm_designator")
+
+    # With the empty_arm_designator locked, it will ALWAYS resolve to the same arm (first resolve is cached), unless it is unlocked
+    # For this challenge, unlocking is not needed
+    self.bag_arm_designator = self.empty_arm_designator.lockable()
+    self.bag_arm_designator.lock()
+
+    # We don't actually grab something, so there is no need for an actual thing to grab
```

Comment:
So why do we need an current_item? Plus why create a variable designator with an entity, when there is also an entitydesignator?